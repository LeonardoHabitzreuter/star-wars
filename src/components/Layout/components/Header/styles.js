import styled, { css } from 'styled-components'

export const StyledHeader = styled.header`
  height: 170px;
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  grid-area: Header;

  ${({ theme: { sizes, colors } }) => css`
    background-color: ${colors.primary};
   
    @media (max-width: ${sizes.phone}) {
      height: 90px;
    }
  `}
`

export const Logo = styled.a`
  background-size: 100%;
  background-repeat: no-repeat;
  width: 293px;
  height: 127px;
  background-image: url('https://static-mh.content.disney.io/starwars/assets/navigation/sw_logo_stacked-336c62367939.png');

  ${({ theme: { sizes } }) => css`
    @media (max-width: ${sizes.phone}) {
      width: 235px;
      height: 25px;
      background-image: url('https://static-mh.content.disney.io/starwars/assets/navigation/sw_logo_horiz@2x-f98247cb30aa.png');
    }
  `}
`
