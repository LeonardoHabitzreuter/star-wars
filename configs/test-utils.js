import * as theme from '../src/styles/theme'
import React from 'react'
import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(
  ui,
  { wrapper: AllTheProviders, ...options }
)

export * from '@testing-library/react'

export { customRender as render }
