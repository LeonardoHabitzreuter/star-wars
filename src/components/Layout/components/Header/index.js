import React from 'react'
import { StyledHeader, Logo } from './styles'

const Header = () => (
  <StyledHeader>
    <Logo alt='logo' href='/' />
  </StyledHeader>
)

export default Header
