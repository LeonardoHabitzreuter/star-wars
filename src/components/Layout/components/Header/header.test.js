import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import React from 'react'
import Header from './index'
import { render } from 'test-utils'

describe('Header component', () => {
  test('should render properly', () => {
    const { container } = render(<Header />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
