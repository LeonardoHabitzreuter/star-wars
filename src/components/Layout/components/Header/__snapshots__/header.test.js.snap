// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render properly 1`] = `
.c0 {
  background-color: #141414;
  height: 170px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: border-box;
  grid-area: Header;
}

.c1 {
  background-size: contain;
  background-repeat: no-repeat;
  width: 293px;
  height: 100px;
  background-image: url('https://static-mh.content.disney.io/starwars/assets/navigation/sw_logo_stacked-336c62367939.png');
}

@media (max-width:414px) {
  .c0 {
    height: 90px;
  }
}

@media (max-width:414px) {
  .c1 {
    width: 235px;
    height: 25px;
    background-image: url('https://static-mh.content.disney.io/starwars/assets/navigation/sw_logo_horiz@2x-f98247cb30aa.png');
  }
}

<header
  class="c0"
>
  <a
    alt="logo"
    class="c1"
    href="/"
  />
</header>
`;
